pkgs/build-support/fetchurl/default.nix

```diff
@@ -132,6 +132,13 @@ let
     else throw "fetchurl requires a hash for fixed-output derivation: ${lib.concatStringsSep ", " urls_}";
 in
 
+assert (lib.isList curlOpts) -> lib.warn ''
+    fetchurl for ${toString (builtins.head urls_)}: curlOpts is a list (${lib.generators.toPretty { multiline = false; } curlOpts}), which is not supported anymore.
+    - If you wish to get the same effect as before, for elements with spaces (even if escaped) to expand to multiple curl arguments, use a string argument instead:
+      curlOpts = ${lib.strings.escapeNixString (toString curlOpts)};
+    - If you wish for each list element to be passed as a separate curl argument, allowing arguments to contain spaces, use curlOptsList instead:
+      curlOptsList = [ ${lib.concatMapStringsSep " " lib.strings.escapeNixString curlOpts} ];'' true;
```

Comment:
... what is the benefit of moving this away from its point of use? The `assert` makes it look like it's a hard check, but it's actually `assert true`, and it's only a warning. What was wrong with the old code, is this worth a refactor?

Reply:
Yes, it is just `assert true`, but it is much more efficient this way because it does not create a thunk out of it.

Reply:
Good point, I didn't see that, thanks!